examples/simple-game-server/Dockerfile.windows

```
@@ -16,16 +16,25 @@
 # Use `--build-arg WINDOWS_VERSION=` to select the correct base image
 # See https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility
 # for details.
+
+# Build Stage
 ARG WINDOWS_VERSION=ltsc2019
 
+FROM --platform=linux/amd64 golang:1.20.4 as base
+
+WORKDIR /gopath/src
+COPY . /gopath/src/
```

Let's make this the same as https://github.com/googleforgames/agones/blob/ae92696f084d7c73d77cbe73bab1a4dff81e892a/examples/simple-game-server/Dockerfile#L16-L21 for consistency.

```suggestion
WORKDIR /go/src
COPY . /go/src/
```

---

The Linux build is functioning properly, but the Windows build has an issue: there are no files in `/go/src/agones.dev/agones/examples/simple-game-server`. This is due to the `COPY . agones.dev/agones` command, which only copies files from the simple-game-server. Log details can be found [here](https://pantheon.corp.google.com/cloud-build/builds;region=global/c4e9f162-e19f-41cd-99b1-f3ceaa3c600f?mods=logs_tg_prod&project=agones-mangalpalli)

```
Step #2 - "push": #9 [base 2/8] WORKDIR /go/src
Step #2 - "push": #9 DONE 0.7s
Step #2 - "push": 
Step #2 - "push": #10 [base 3/8] COPY . agones.dev/agones
Step #2 - "push": #10 DONE 0.1s
Step #2 - "push": 
Step #2 - "push": #11 [base 4/8] RUN ls -ltr
Step #2 - "push": #0 0.150 total 4
Step #2 - "push": #0 0.150 drwxr-xr-x 3 root root 4096 Aug 14 19:19 agones.dev
Step #2 - "push": #11 DONE 0.2s
Step #2 - "push": 
Step #2 - "push": 
Step #2 - "push": #12 [base 5/8] RUN ls -ltr /go/src/agones.dev/agones
Step #2 - "push": #0 0.135 total 88
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1129 Aug  4 01:08 gameserverallocation.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1080 Aug  4 01:08 gameserver.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1183 Aug  4 01:08 gameserver-windows.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1131 Aug  4 01:08 gameserver-passthrough.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root   945 Aug  4 01:08 fleetautoscaler.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1136 Aug  4 01:08 fleet.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1406 Aug  4 01:08 fleet-tcp.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1346 Aug  4 01:08 fleet-distributed.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1200 Aug  4 01:08 dev-gameserver.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  3682 Aug  4 01:08 README.md
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1017 Aug  4 01:08 Dockerfile
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  2507 Aug  7 17:13 go.sum
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root   578 Aug  7 17:13 go.mod
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root 16836 Aug 10 13:15 main.go
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1300 Aug 12 19:22 cloudbuild.yaml
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  5481 Aug 12 19:22 Makefile
Step #2 - "push": #0 0.135 -rw-r--r-- 1 root root  1661 Aug 14 19:14 Dockerfile.windows
Step #2 - "push": #12 DONE 0.2s
Step #2 - "push": 
Step #2 - "push": #13 [base 6/8] WORKDIR /go/src/agones.dev/agones/examples/simple-game-server
Step #2 - "push": #13 DONE 0.0s
Step #2 - "push": 
Step #2 - "push": #14 [base 7/8] RUN ls -ltr
Step #2 - "push": #14 0.153 total 0
Step #2 - "push": #14 DONE 0.2s
```

---

🤔  so the line that fails is:

```
Step #2 - "push": #15 0.339 agones.dev/agones@v1.33.0 (replaced by ../../): reading /go/src/go.mod: open /go/src/go.mod: no such file or directory
```

I think I see the issue, it's in the makefile.

You can see how the linux image is built to run from the `$(root_path)`:
https://github.com/googleforgames/agones/blob/2a6091ff43e4d6bfcc221b5e7ed05ba978e1d223/examples/simple-game-server/Makefile#L125-L128

But if we look at the Windows target, we see:
https://github.com/googleforgames/agones/blob/2a6091ff43e4d6bfcc221b5e7ed05ba978e1d223/examples/simple-game-server/Makefile#L111C1-L114C141

Where previously there should be `.` there is `examples/simple-game-server/`. I think if you fixup that, then that will resolve the issue (or at least get you closer).